### PR TITLE
UML-2671: Show metrics to accounts in stats page

### PR DESCRIPTION
**Pull request by SeemaMenon**
# Purpose

https://opgtransform.atlassian.net/browse/UML-2671

Fixes UML-2671

## Approach

1. Introduced new handler for metrics page
2. Introduced new data handler to query metrics as a batch and handle the responses [to return a map of records for the last three months time period]
3. New UI page
4. Tests

## Learning

_Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about the Use a Lasting Power of Attorney service_

## Checklist

* [X] I have performed a self-review of my own code
* [X] I have added relevant logging with appropriate levels to my code
~New event_codes have been documented on the [wiki page]~(https://opgtransform.atlassian.net/wiki/spaces/LSML2/pages/3277881441/Understanding+the+event+logs)
~I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant~
* [X] I have added tests to prove my work
~I have added welsh translation tags and updated translation files~
~I have run an accessibility tool on any pages I have made changes to and fixed any issues found~
~I have notified the Interaction Designer of any content changes so that appropriate screenshots/flow diagram changes can be made~
* [ ] The product team have tested these changes

### CODE REVIEW

**Comment by SeemaMenon** [comment]
Screenshot 2022-11-16 at 16 38 04

**Comment by SeemaMenon** [comment]
Report page design updated to now have aggregated blocks in individual tables [as per Sarah]

Screenshot 2022-11-17 at 12 35 52

File: service-admin/internal/server/data/statistics_test.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+       "Total": {
+          "lpas_added": 12,
+          "lpa_removed_evet": 12,
+          "account_created_event": 15,
+        },
+      },
+      wantErr: assert.NoError, errorExpected: false,
+    },
+    {
+			name:  "doesn't find a metric for time period",
```
Comment (cooperaj) [inline service-admin/internal/server/data/statistics_test.go:103]: This indentation seems screwy?

File: service-admin/internal/server/handlers/stats.go [hunk 2]
```diff
@@ -3,26 +3,65 @@ package handlers
...
+		r, err := s.statisticsService.GetAllMetrics(ctx, checkList)
+		if err != nil {
+		return nil
+		}
+	
+	return r
+}
+
+func getPeriodsToBeChecked() []string {
+		currentTime := time.Now()
```
Comment (cooperaj) [inline service-admin/internal/server/handlers/stats.go:53]: As does this. Is gofmt running?

File: service-admin/internal/server/handlers/stats_test.go [hunk 3]
```diff
@@ -7,13 +7,125 @@ import (
 	"net/http/httptest"
 	"strings"
 	"errors"
+	"reflect"
 
 	. "github.com/ministryofjustice/opg-use-an-lpa/service-admin/internal/server/handlers"
 )
 
+type mockStatisticsService struct {
+	GetAllMetricsFunc func(context.Context, []string) (map[string]map[string]float64, error)
```
Comment (cooperaj) [inline service-admin/internal/server/handlers/stats_test.go:16]: Seems to have switched to tabs for indentation rather than spaces

**Review by cooperaj — APPROVED** [review]

**Pull request merged**